src/common/logging/Logging.c

```diff
@@ -53,6 +53,7 @@ void SetConsoleLoggingEnabled(bool enabledOrDisabled)
 const char* GetLoggingLevelName(LoggingLevel level)
 {
     const char* result = g_debug;
+    (void)result;
```

Comment:
>(void)result; [](http://example.com/codeflow?start=4&length=13)

Line 90 ensures that result is always set after the initialization on line 55. Could you please summarize why is it necessary?

Reply:
This gets flagged as dead store, since the value assigned to `result` in line 55 always gets overwritten by all branches (including `default`) in the `switch` statement. But removing the initialization here would be in conflict with our requirement to always initialize variables.

I would agree that this is a false-positive, that's why I added the `(void)result` (it suppresses this warning https://clang.llvm.org/docs/analyzer/user-docs/FAQ.html#dead-store).

Other alternatives didn't work well: removing the `default` label changes nothing as there are cases for all possible enum values, variable initializer still gets overwritten.